tricycle_controller/src/tricycle_controller.cpp

```diff
@@ -392,6 +382,11 @@ CallbackReturn TricycleController::on_activate(const rclcpp_lifecycle::State &)
 CallbackReturn TricycleController::on_deactivate(const rclcpp_lifecycle::State &)
 {
   subscriber_is_active_ = false;
+  if (!is_halted)
+  {
+    halt();
+    is_halted = true;
+  }
```

Comment:
@christophfroehlich why do we need this here?

Reply:
to stop the vehicle in on_deactivate transition

Reply:
Same as above, can't we simply call halt without any check in the deactivate method?